### PR TITLE
log message with the time of the item

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco key move here is having `logMessage` use the ctime from the event if it gets sent in, this is useful for testing.

### CODE REVIEW

File: rpc/client/client.go [hunk 1]
```diff
@@ -156,3 +156,7 @@ func (c *Client) StateMachineLatestCTime() *time.Time {
 func (c *Client) StateMachineInBandMessagesSince(t time.Time) ([]gregor.InBandMessage, error) {
 	return c.sm.InBandMessagesSince(c.user, c.device, t)
 }
+
+func (c *Client) StateMachineState(t gregor.TimeOrOffset) (gregor.State, error) {
+	return c.sm.State(c.user, c.device, t)
+}
```
Comment (maxtaco) [inline rpc/client/client.go]: and what's this?

File: storage/mem_sm.go [hunk 2]
```diff
@@ -252,12 +252,14 @@ func (m *MemEngine) consumeInBandMessage(uid gregor.UID, msg gregor.InBandMessag
...
 	default:
 	}
-	user.logMessage(now, msg, i)
 
 	retTime := now
 	if i != nil {
 		retTime = i.ctime
 	}
+
+	user.logMessage(retTime, msg, i)
```
Comment (maxtaco) [inline storage/mem_sm.go]: got it 👍

**Comment by maxtaco** [comment]
had a brief q

File: rpc/client/client.go (hunk 1 above)
Comment (mmaxim) [inline rpc/client/client.go]: It was convenient to have for my tests in my keybase client PR

Comment (maxtaco) [inline rpc/client/client.go]: Ok thumbs up!

**Pull request merged**